qiskit/circuit/library/__init__.py

```diff
@@ -10,38 +10,121 @@
 
-   from qiskit.circuit.library import MCXGate
-   gate = MCXGate(4)
+   from qiskit.circuit import QuantumCircuit
+   from qiskit.circuit.library import PauliEvolutionGate
+   from qiskit.quantum_info import SparsePauliOp
 
-   from qiskit import QuantumCircuit
-   circuit = QuantumCircuit(5)
-   circuit.append(gate, [0, 1, 4, 2, 3])
-   circuit.draw('mpl')
+   hamiltonian = SparsePauliOp(["ZZI", "IZZ", "IXI"], coeffs=[1, 1, -1])
+   gate = PauliEvolutionGate(hamiltonian)
 
-The library is organized in several sections. The function
-:func:`.get_standard_gate_name_mapping` allows you to see the available standard gates and operations.
+   circuit = QuantumCircuit(hamiltonian.num_qubits)
+   circuit.append(gate, circuit.qubits)
 
-.. autofunction:: get_standard_gate_name_mapping
+   circuit.draw("mpl")
+
+This library is organized in different sections:
+
+   * :ref:`Standard gates <standard-gates>`, :ref:`directives <standard-directives>`,
+     and :ref:`operations <standard-operations>`
+   * :ref:`Generalized gates <generalized-gates>`
+   * :ref:`Arithmetic operations <arithmetic>`
+   * :ref:`Basis changes <basis-change>`
+   * :ref:`Boolean logic <boolean-logic>`
+   * :ref:`Data encoding <data-encoding>` and :ref:`preparation <data-preparation>`
+   * :ref:`Particular operations <particular>`
+   * :ref:`N-local circuits <n-local>`
+   * :ref:`Oracles <oracles>`
+   * :ref:`Template circuits <template>`
+
+We distinguish into different categories of operations:
+
+Standard gates
+   These are fundamental quantum gates, a subset of which typically forms a basis gate
+   set on a quantum computer. These are unitary operations represented as :class:`.Gate`.
+   The library also provides standard compiler directives (a :class:`.Barrier`) and non-unitary
+   operations (like :class:`.Measure`).
+
+Abstract operations
+   This describes operations that are defined by a mathematical action, but can be implemented with
+   different decompositions. For example, a multi-controlled X gate flips the target qubit if all
+   control qubits are :math:`|1\rangle`, and there are a variety of concrete circuits implementing
+   this operation using lower-level gates. Such abstract operations are represented as :class:`.Gate`
+   or  :class:`~.circuit.Instruction`. This allows building the circuit without choosing a concrete
+   implementation of each block and, finally, let the compiler (or you as user) choose the optimal
+   decomposition. For example:
+
+   .. plot::
+      :alt: A circuit with a multi-controlled X gate.
+      :include-source:
+
+      from qiskit.circuit.library import MCXGate
+      mcx = MCXGate(4)
+
+      from qiskit import QuantumCircuit
+      circuit = QuantumCircuit(5)
+      circuit.append(mcx, [0, 1, 4, 2, 3])
+      circuit.draw("mpl")
+
+   When this circuit is transpiled the circuit context is taken into account. For example, if idle
+   qubits are available they can be used to obtain a shallower circuit::
```

Comment:
```suggestion
   For circuits with abstract operations, the circuit context is taken into account during transpilation. For example, if idle
   qubits are available, they can be used to obtain a shallower circuit::
```

Reply:
"when this circuit is transpiled" sounds a bit like "after transpilation", but it's during.